error: [Prusti: verification error] postcondition might not hold.
  --> $DIR/failures.rs:12:11
   |
12 | #[ensures(false)]
   |           ^^^^^
   |
note: the error originates here
  --> $DIR/failures.rs:13:1
   |
13 | / fn failing_postcondition(x: i32) {
14 | |     let mut x = x;
15 | |     while x < 10 {
16 | |         x += 1;
17 | |     }
18 | | }
   | |_^

error: [Prusti: verification error] loop invariant might not hold in the first loop iteration.
  --> $DIR/failures.rs:23:25
   |
23 |         body_invariant!(false);
   |                         ^^^^^
   |
note: the error originates here
  --> $DIR/failures.rs:23:25
   |
23 |         body_invariant!(false);
   |                         ^^^^^

error: [Prusti: verification error] precondition might not hold.
  --> $DIR/failures.rs:29:5
   |
29 |     failing_precondition(42);
   |     ^^^^^^^^^^^^^^^^^^^^^^^^
   |
note: the failing assertion is here
  --> $DIR/failures.rs:4:12
   |
4  | #[requires(false)]
   |            ^^^^^

error: aborting due to 3 previous errors

